File: elixir/emit_logs.exs
# create connection
{:ok, connection} = AMQP.Connection.open
{:ok, channel} = AMQP.Channel.open(connection)

# get message
message =
  case System.argv do
    []    -> "Elixir log message..."
    words -> Enum.join(words, " ")
  end

# create exchange
AMQP.Exchange.declare(channel, "logs", :fanout)

# push a log message
AMQP.Basic.publish(channel, "logs", "", message)
IO.puts " [x] Published '#{message}'"

AMQP.Connection.close(connection)

File: ruby/emit_logs.rb
#!/usr/bin/env ruby
require 'bunny'

# create connection
conn = Bunny.new
conn.start

# create channel
ch = conn.create_channel

# define exchange
x = ch.fanout 'logs'

# get message
msg = ARGV.empty? ? 'Ruby log message...' : ARGV.join(' ')

# publish log to the exchange
x.publish(msg)
puts " [x] Sent #{msg}"

# close a connection
conn.close
